src/operators/reduceState.spec.ts

```diff
@@ -180,3 +180,47 @@ test(
     m.expect(routine$, sub2).toBeObservable(sub2Expected$);
   })
 );
+
+test(
+  'reduceState emits updated value to derived streams if routine subscribes first',
+  marbles(m => {
+    const defaultState = 1;
+    const action$ = m.hot('      2--(12)-2-', actions);
+    const sub1 = '               -^--------';
+    const sub1Expected$ = m.hot('-1-2------', numbers);
+    const sub2 = '               ^---------';
+    const sub2Expected$ = m.hot('1--2----2-', numbers);
+
+    const state$ = action$.pipe(
+      reduceState('testStream', defaultState, [reducers.handleOne])
+    );
+    const routine$ = action$.pipe(
+      ofType(incrementMocks.actionCreators.incrementMany),
+      withLatestFrom(state$),
+      map(([, s]) => s)
+    );
+
+    m.expect(state$, sub1).toBeObservable(sub1Expected$);
+    m.expect(routine$, sub2).toBeObservable(sub2Expected$);
+  })
+);
+
+test.failing(
+  'reduceState should only emit the latest state processed at the current frame to prevent glitches',
```

Comment:
I think we should discuss the expectation here. This is not really a glitch, it's just two iterations in one frame.

Reply:
True, I guess my intent behind the wording is that two iterations in one frame can lead to glitches further down the graph